src/ILCompiler.Compiler/src/Compiler/SingleFileCompilationModuleGroup.cs

```diff
@@ -27,7 +27,11 @@ public override void AddCompilationRoots(IRootingServiceProvider rootProvider)
         {
             base.AddCompilationRoots(rootProvider);
 
-            AddCompilationRootsForExports((EcmaModule)_typeSystemContext.SystemModule, rootProvider);
+            foreach (var inputFile in _typeSystemContext.ReferenceFilePaths)
+            {
```

Comment:
Also note that we'll have to restore special-case handling for `System.Private.CoreLib` if we go this route.